classes/WP_CLI/Action/Cancel.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+use function \WP_CLI\Utils\get_flag_value;
+
+/**
+ * WP-CLI command: action-scheduler action cancel
+ */
+class ActionScheduler_WPCLI_Action_Cancel_Command extends ActionScheduler_WPCLI_Command {
+
+	/**
+	 * Execute command.
+	 *
+	 * @uses as_unschedule_action()
+	 * @uses as_unschedule_all_actions()
+	 * @uses $this->print_error()
+	 * @uses $this->print_success()
+	 * @return void
+	 */
+	public function execute() {
+		$hook          = $this->args[0];
+		$group         = null;
+		$callback_args = get_flag_value( $this->assoc_args, 'args', null );
+		$all           = get_flag_value( $this->assoc_args, 'all' );
+
+		if ( ! empty( $this->args[1] ) ) {
+			$group = $this->args[1];
+		}
+
+		if ( ! empty( $callback_args ) ) {
+			$callback_args = json_decode( $callback_args, true );
+		}
+
+		$function_name = 'as_unschedule_action';
+		$multiple      = false;
+
+		if ( $all ) {
+			$function_name = 'as_unschedule_all_actions';
+			$multiple      = true;
+		}
+
+		try {
+			call_user_func( $function_name, $hook, $callback_args, $group );
+		} catch ( \Exception $e ) {
+			$this->print_error( $e, $multiple );
```

Comment:
note that it won't throw an exception necessarily even if cancelling failed, there could be an error caught by the try_catch in `as_unschedule_action`. It would be more reliable to either 
1. check the logs for failed action, or
2. verify that the action is indeed cancelled, or
3. modify the `as_unschedule_action` to return the result value so that we can display it reliably.

Reply:
Going to rework this command a bit: realized in its current state actions could not be unscheduled by group only.